### ISSUE TITLE
Selection behaviour when formatted text is selected

### ISSUE THREAD

**Issue by becdetat** [Code52/DownmarkerWPF#265]
If formatted text is selected (a block beginning and ending with _, *_ or ***), for example

```
this *is a* test
```

with `*is a*` selected.
- pressing left should move to between `*` and `is a`
- pressing right should move to between `is a` and `*`
- pressing any non-control key to insert text (inc CTRL-C) should replace `is a` but leave the formatting marks
- pressing backspace should delete the entire selection (but delete should just delete the content)
- pressing ctrl-i and ctrl-b work as they currently do now

In other words, when a formatted block is selected, MarkPad should act as if only the enclosed text is selected, ignoring the formatting, except for when pressing backspace. This should match the behaviour of Word for example.
